src/main/java/org/apache/sling/pipes/internal/PipeBuilderImpl.java

```diff
@@ -376,6 +377,17 @@ public Job runAsync(Map bindings) throws PersistenceException {
         return plumber.executeAsync(resolver, pipe.getResource().getPath(), bindings);
     }
 
+    @Override
+    public ExecutionResult runParallel(int numThreads) throws Exception {
+        JsonWriter writer = new JsonWriter();
+        writer.starts();
+        Pipe pipe = this.build();
```

Comment:
this looks strange, i wonder if we shouldn't make that call just change the main pipe from container to "threadedpipe"